web/server/codechecker_server/permissions.py

```diff
@@ -293,13 +293,16 @@ def __get_perm_record(self, auth_name, is_group):
                 query(SysPerm). \
                 filter(and_(
                     SysPerm.permission == self._perm_name,
-                    SysPerm.name == auth_name,
+                    func.lower(SysPerm.name) == auth_name.lower(),
                     SysPerm.is_group == is_group
                 )).one_or_none()
 
             return record
 
         def _add_perm_impl(self, auth_name, is_group=False):
+            if not auth_name or len(auth_name) <= 0:
```

Comment:
How is it possible that the length of a string is less than `0`? The zero case is covered in the first expression in this if (`not auth_name`).

Reply:
Done.